ci/docker/cuda-10.0-cpp.dockerfile

```diff
@@ -59,11 +59,15 @@ RUN apt-get update -y -q && \
       ninja-build \
       pkg-config \
       protobuf-compiler \
+      python-minimal \
       rapidjson-dev \
       thrift-compiler \
       tzdata && \
       apt-get clean && rm -rf /var/lib/apt/lists*
 
+# Make sure libcuda.so and its dependencies are visible by CMake
+RUN ln -s /usr/local/cuda-10.0/compat/lib* /usr/local/cuda-10.0/lib64/
```

Comment:
It seems that we need to use `/usr/local/cuda/lib64/stubs/libcuda.so` instead of `/usr/local/cuda-10.0/compat/lib/libcuda.so`: https://github.com/NVIDIA/nvidia-docker/issues/76#issuecomment-210635768

I'll look info how to use `stubs/libcuda.so`.

Reply:
How about this?

```diff
diff --git a/ci/docker/cuda-10.0-cpp.dockerfile b/ci/docker/cuda-10.0-cpp.dockerfile
index 9df105b7a..3a706a1f2 100644
--- a/ci/docker/cuda-10.0-cpp.dockerfile
+++ b/ci/docker/cuda-10.0-cpp.dockerfile
@@ -65,9 +65,6 @@ RUN apt-get update -y -q && \
       tzdata && \
       apt-get clean && rm -rf /var/lib/apt/lists*
 
-# Make sure libcuda.so and its dependencies are visible by CMake
-RUN ln -s /usr/local/cuda-10.0/compat/lib* /usr/local/cuda-10.0/lib64/
-
 # Prioritize system packages and local installation
 # The following dependencies will be downloaded due to missing/invalid packages
 # provided by the distribution:
@@ -90,6 +87,7 @@ ENV ARROW_BUILD_STATIC=OFF \
     ARROW_NO_DEPRECATED_API=ON \
     ARROW_PLASMA=ON \
     ARROW_USE_CCACHE=ON \
+    CUDA_LIB_PATH=${LIBRARY_PATH} \
     GTest_SOURCE=BUNDLED \
     ORC_SOURCE=BUNDLED \
     PATH=/usr/lib/ccache/:$PATH \
```

Reply:
It seems to compile fine indeed. As for runtime tests, I can't tell you: I don't have a CUDA 10.0 install (only 9.1).

Reply:
Oh. I only have CUDA 10.1. I confirmed that this change works with `nvidia/cuda:10.1-devel-ubuntu18.04` base image.

I'll merge this and add support for CUDA 10.1 as a follow-up task.